web/packages/teleport/src/services/mfa/mfaOptions.ts

```diff
@@ -0,0 +1,45 @@
+import { Auth2faType } from 'shared/services';
+
+import { DeviceType, MfaAuthenticateChallenge } from './types';
+
+export type MfaOption = {
+  value: DeviceType;
+  label: string;
+};
+
+export function getMfaChallengeOptions(mfaChallenge: MfaAuthenticateChallenge) {
+  const mfaOptions: MfaOption[] = [];
+
+  if (mfaChallenge?.webauthnPublicKey) {
+    mfaOptions.push({ value: 'webauthn', label: 'Passkey or Security Key' });
+  }
+
+  if (mfaChallenge?.totpChallenge) {
+    mfaOptions.push({ value: 'totp', label: 'Authenticator App' });
+  }
+
+  if (mfaChallenge?.ssoChallenge) {
+    mfaOptions.push({
+      value: 'sso',
+      label:
+        mfaChallenge.ssoChallenge.device.displayName ||
+        mfaChallenge.ssoChallenge.device.connectorId,
+    });
+  }
+
+  return mfaOptions;
+}
+
+export function getMfaRegisterOptions(auth2faType: Auth2faType) {
+  const mfaOptions: MfaOption[] = [];
+
+  if (auth2faType === 'webauthn' || auth2faType === 'on') {
+    mfaOptions.push({ value: 'webauthn', label: 'Passkey or Security Key' });
+  }
+
+  if (auth2faType === 'otp' || auth2faType === 'on') {
+    mfaOptions.push({ value: 'totp', label: 'Authenticator App' });
```

Comment:
Seems like these objects can be defined outside of the functions and then it's just `mfaOptions.push(WEBAUTHN_OPTION)` in both methods

Reply:
Good point, added `const webauthnOption`, `const totpOption`, and `func getSsoOption`.

SSO option is derived from the challenge so I'm not sure if an enum works here (that's generally what all caps like `WEBAUTHN_OPTION` is for right?).

Reply:
I normally use `ALL_CAPS` for constants

Reply:
done